tests/libs/datasets/new_cases_and_deaths_test.py

```diff
@@ -154,6 +154,38 @@ def test_spread_first_reported_value():
     pd.testing.assert_series_equal(expected, results)
 
 
+def test_spread_first_reported_value_2():
+    series = test_helpers.series_with_date_index([None, None, 10, 5, 0, 0, 0, 12, 0, 4])
+
+    results = new_cases_and_deaths.spread_first_reported_value_after_stall(series)
+
+    expected = test_helpers.series_with_date_index([None, None, 10, 5, 3, 3, 3, 3, 2, 2])
+
+    pd.testing.assert_series_equal(expected, results)
+
+
+def test_spread_first_reported_value_leading_zeros():
+    series = test_helpers.series_with_date_index([0, 0, 10, 5, 0, 0, 0, 12, 5])
+
+    results = new_cases_and_deaths.spread_first_reported_value_after_stall(series)
+
+    # We don't spread across leading zeros.
+    expected = test_helpers.series_with_date_index([0.0, 0, 10, 5, 3, 3, 3, 3, 5])
+
+    pd.testing.assert_series_equal(expected, results)
+
+
+def test_spread_first_reported_value_trailing_zeros():
+    series = test_helpers.series_with_date_index([5, 0, 0, 0, 12, 0, 0])
+
+    results = new_cases_and_deaths.spread_first_reported_value_after_stall(series)
+
+    # We don't spread across leading zeros.
+    expected = test_helpers.series_with_date_index([5, 3, 3, 3, 3, 0, 0])
```

Comment:
This test fails because the zeros turn into `NaN`.  Maybe that's okay?  I'm not sure what we want / if it matters.

Reply:
I think we should keep the trailing zeros - we have other processes that handle that (see stalled tail filter (I think thats the name of it)).

I'm still keeping the hop alive that there will be real days of zeros! :D